labgraph/lsl_node/tests/test_lsl_node.py

```diff
@@ -0,0 +1,253 @@
+from pylsl import StreamInfo, StreamOutlet, StreamInlet, resolve_stream
+import pytest
+import asyncio
+import time
+from multiprocessing import Process
+from typing import Sequence
+
+from ...graphs import (
+    AsyncPublisher,
+    Config,
+    Connections,
+    Graph,
+    Module,
+    Node,
+    Topic,
+    publisher,
+    subscriber
+)
+
+from ...runners import LocalRunner, NormalTermination, ParallelRunner
+from ...util.testing import get_test_filename, local_test
+from ..lsl_message import LSLMessage
+from ..lsl_poller_node import LSLPollerConfig, LSLPollerNode
+from ..lsl_sender_node import LSLSenderConfig, LSLSenderNode
+
+NUM_MESSAGES = 10
+SAMPLE_RATE = 10
+
+DATA_DELIMITER = "\n"
+samples = [0.1, 1.1, 2.3, 4.4, 5.5, 6.6, 7.5, 7.7, 8.8, 9.9]
+print(f' original {samples}')
+
+
+class MySinkConfig(Config):
+    output_filename: str = get_test_filename()
+
+
+class MySink(Node):
+    """
+    Convenience node for receiving messages from a `LSLSenderNode`.
+    """
+
+    TOPIC = Topic(LSLMessage)
+    config: MySinkConfig
+
+    def setup(self) -> None:
+        self.output_file = open(self.config.output_filename, "w+")
+        self.num_received = 1
+
+    @subscriber(TOPIC)
+    async def sink(self, message: LSLMessage) -> None:
+        for value in message.data:
+            value_str = str(value)
+            self.output_file.write(value_str)
+            self.output_file.write(DATA_DELIMITER)
+        self.num_received += 1
+
+        if self.num_received == NUM_MESSAGES:
+            raise NormalTermination()
+
+    def cleanup(self) -> None:
+        self.output_file.close()
+
+
+class MySourceConfig(Config):
+    should_terminate: bool = True
+
+
+class MySource(Node):
+    """
+    Convenience node for supplying messages to a LSLSenderNode
+    """
+    TOPIC = Topic(LSLMessage)
+    config: MySourceConfig
+
+    @publisher(TOPIC)
+    async def source(self) -> AsyncPublisher:
+
+        yield self.TOPIC, LSLMessage(samples)
+        await asyncio.sleep(1/SAMPLE_RATE)
+
+        if self.config.should_terminate:
+            raise NormalTermination()
+
+
+# send messages
+def write_sample_to_lsl() -> None:
+    srate = 100
+    name = 'Mock LSL Sender'
+    type = 'EEG'
```

Comment:
Please use "Mock_Signal" instead of a specific signal name.

Reply:
Done

Reply:
Please update for all instances.